scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala

```diff
@@ -0,0 +1,387 @@
+    * The object shall never be used after it is disposed.
+    */
+  def dispose() : Unit = nd.dispose()
+
+  /**
+    * Dispose all NDArrays who help to construct this array. <br />
+    * e.g. (a * b + c).disposeDeps() will dispose a, b, c (including their deps) and a * b
+    * @return this array
+    */
+  def disposeDeps() : NDArray = nd.disposeDepsExcept()
+  // def disposeDepsExcept(arr : Array[NDArray]) : NDArray = nd.disposeDepsExcept()
+
+  /**
+    * Return a sliced NDArray that shares memory with current one.
+    * NDArray only support continuous slicing on axis 0
+    *
+    * @param start Starting index of slice.
+    * @param stop Finishing index of slice.
+    *
+    * @return a sliced NDArray that shares memory with current one.
+    */
+  def slice(start : Int, stop : Int) : NDArray = nd.slice(start, stop)
+
+  /**
+    * Return a sliced NDArray at the ith position of axis0
+    * @param i
+    * @return a sliced NDArray that shares memory with current one.
+    */
+  def slice (i : Int) : NDArray = nd.slice(i)
+
+  /**
+    * Return a sub NDArray that shares memory with current one.
+    * the first axis will be rolled up, which causes its shape different from slice(i, i+1)
+    * @param idx index of sub array.
+    */
+  def at(idx : Int) : NDArray = nd.at(idx)
+
+  def T : NDArray = nd.T
+
+  /**
+    * Get data type of current NDArray.
+    * @return class representing type of current ndarray
+    */
+  def dtype : DType = nd.dtype
+
+  /**
+    * Return a copied numpy array of current array with specified type.
+    * @param dtype Desired type of result array.
+    * @return A copy of array content.
+    */
+  def asType(dtype : DType) : NDArray = nd.asType(dtype)
+
+  /**
+    * Return a reshaped NDArray that shares memory with current one.
+    * @param dims New shape.
+    *
+    * @return a reshaped NDArray that shares memory with current one.
+    */
+  def reshape(dims : Array[Int]) : NDArray = nd.reshape(dims)
+
+  /**
+    * Block until all pending writes operations on current NDArray are finished.
+    * This function will return when all the pending writes to the current
+    * NDArray finishes. There can still be pending read going on when the
+    * function returns.
+    */
+  def waitToRead(): Unit = nd.waitToRead()
+
+  /**
+    * Get context of current NDArray.
+    * @return The context of current NDArray.
+    */
+  def context : Context = nd.context
+
+  /**
+    * Set the values of the NDArray
+    * @param value Value to set
+    * @return Current NDArray
+    */
+  def set(value : Float) : NDArray = nd.set(value)
+  def set(other : NDArray) : NDArray = nd.set(other)
+  def set(other : Array[Float]) : NDArray = nd.set(other)
+
+  def add(other : NDArray) : NDArray = this.nd + other.nd
+  def add(other : Float) : NDArray = this.nd + other
+  def _add(other : NDArray) : NDArray = this.nd += other
+  def _add(other : Float) : NDArray = this.nd += other
```

Comment:
underline usually means private / internal functions, and the name itself does not show the functionality. consider name it `addInplace`.